scrypto/src/resource/auth_zone.rs

```diff
@@ -15,13 +15,13 @@ pub trait ScryptoAuthZone {
         &self,
         amount: A,
         resource_address: ResourceAddress,
-    ) -> Proof;
+    ) -> FungibleProof;
```

Comment:
Is this true? Is a proof created by amount always fungible? IIRC I don't think that this is correct.

Reply:
> According to the sources one can create_proof_of_amount of non-fungibles if amount is None (see [compose_proof_by_amount](https://github.com/radixdlt/radixdlt-scrypto/blob/ea3d7f1f1ef56de3e03c44485f0f21efb113b82f/radix-engine/src/blueprints/resource/auth_zone/auth_zone_composition.rs#L78))

I was wrong, [compose_proof_by_amount](https://github.com/radixdlt/radixdlt-scrypto/blob/ea3d7f1f1ef56de3e03c44485f0f21efb113b82f/radix-engine/src/blueprints/resource/auth_zone/auth_zone_composition.rs#L78) is able to return non-fungible proof if amount is `Some`. Will revert this